Platform/TigerlakeBoardPkg/Library/Stage2BoardInitLib/Stage2BoardInitLib.c

```diff
@@ -682,7 +683,16 @@ ClearSmi (
   // Clear the status before setting smi enable
   //
   SmiSts = IoRead32 ((UINTN)(UINT32)(ACPI_BASE_ADDRESS + R_ACPI_IO_SMI_STS));
-  Pm1Sts = IoRead16 ((UINTN)(ACPI_BASE_ADDRESS + R_ACPI_IO_PM1_STS));
+  Pm1Sts = IoRead32 ((UINTN)(ACPI_BASE_ADDRESS + R_ACPI_IO_PM1_STS));
+  Pm1Cnt = IoRead16 ((UINTN)(ACPI_BASE_ADDRESS + R_ACPI_IO_PM1_CNT));
+
+  // Clear RTC Alarm only if wake source is RTC SMI#
+  if(((Pm1Sts & (B_ACPI_IO_PM1_STS_RTC_EN | B_ACPI_IO_PM1_STS_RTC)) ==
+       (BIT26 | B_ACPI_IO_PM1_STS_RTC)) &&
+     !(Pm1Cnt & B_ACPI_IO_PM1_CNT_SCI_EN)) {

```

Comment:
Do we need to check B_ACPI_IO_PM1_CNT_SCI_EN?

Reply:
If PM1_CNT.SCI_EN is SET and PM1_STS indicates there has a RTC alarm, then it means only SCI is generated. I think ClearSmi() should not clear SCI event. That's why PM1_CNT.SCI_EN should be checked to ensure the RTC alarm caused an SMI.

Reply:
Thanks for clarification. Can you make this condition more readable?
```
  if (((Pm1Sts & B_ACPI_IO_PM1_STS_RTC_EN) != 0) &&
      ((Pm1Sts & B_ACPI_IO_PM1_STS_RTC) != 0) &&
      ((Pm1Cnt & B_ACPI_IO_PM1_CNT_SCI_EN) == 0)) {
```